doordeck-sdk/src/mingwMain/resources/doordeck_headless_sdk/utils/Utils.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Text;
+using System.Text.Json;
+
+namespace DoordeckHeadlessSDK.utils
+{
+    public static unsafe class Utils
+    {
+        private static readonly JsonSerializerOptions JsonSerializerOptions = new()
+        {
+            Converters = {
+                new AuthKeyResponseConverter()
+            }
+        };
+
+        public static sbyte* ToData<T>(this T input) =>
+            JsonSerializer.Serialize(input, JsonSerializerOptions).ToSByte();
+
+        public static T? FromData<T>(sbyte* input) =>
+            JsonSerializer.Deserialize<T>(ConvertSByteToString(input), JsonSerializerOptions);
+
+        public static unsafe sbyte* ToSByte(this string input)
+        {
+            byte[] bytes = Encoding.ASCII.GetBytes(input);
+            fixed (byte* po = bytes)
+            {
+                return (sbyte*)po;
+            }
```

Comment:
You maybe able to use https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshal.stringtohglobalansi?view=net-9.0

Reply:
This one works as well!